### PR TITLE
Update brigade.js to use the GitHub library

### PR DESCRIPTION
**Pull request by radu-matei**
Note that this PR does _not_ handle the issue of actually testing itself on each pull request, but just replaces the existing GitHub notification logic with the one in the utils library.

### CODE REVIEW

File: brigade.js [hunk 1]
```diff
@@ -28,166 +17,13 @@ function build(e, project) {
...
-            console.error("original error: " + e.toString());
-            return e2;
-        }
-    }
-}
+events.on("exec", runSuite);
+events.on("check_suite:requested", runSuite);
+events.on("check_suite:rerequested", runSuite);
+events.on("issue_comment:created", (e, p) => Check.handleIssueComment(e, p, runSuite));
+events.on("issue_comment:edied", (e, p) => Check.handleIssueComment(e, p, runSuite));
```
Comment (vdice) [inline brigade.js]: typo on the action here: `s/edied/edited`

Comment (vdice) [reply brigade.js]: We've lost a handler for `events.on("check_run:rerequested", runCheck);`... file an issue to add back in if not in this same PR?

**Review by vdice — COMMENTED** [review]
:clap: Ooooh so streamlined, so clean!  A few comments/questions.

File: brigade.js (hunk 1 above)
Comment (radu-matei) [reply brigade.js]: Opened #21 to track this - as we only have one check run at the moment, I don't see this as a priority, but feel free to disagree, and I'll tackle it next.

**Review by vdice — APPROVED** [review]

**Pull request merged**